src/features/device-state/routes/state.ts

```diff
@@ -25,6 +26,163 @@ export const varListInsert = (varList: EnvVarList, obj: Dictionary<string>) => {
 	});
 };
 
+export type App = {
+	name: string;
+	commit: string;
+	releaseId: number;
+	releaseVersion: string;
+	appId?: string;
+	uuid?: string;
+	services: {
+		[id: number]: AppService;
+	};
+	volumes: any;
+	networks: any;
+};
+
+export type AppService = {
+	imageId: number;
+	serviceName: string;
+	image: string;
+	running: boolean;
+	environment: Dictionary<string>;
+	labels: Dictionary<string>;
+	contract?: string;
+};
+
+export type LocalState = {
+	name: string;
+	config: Dictionary<string>;
+	apps: Dictionary<Partial<App>>;
+};
+
+async function buildAppFromRelease(
+	device: AnyObject,
+	application: AnyObject,
+	release: AnyObject,
+	config: Dictionary<string>,
+): Promise<App> {
+	let composition: any = {};
+	const services: App['services'] = {};
+
+	// Parse the composition to forward values to the device
+	if (_.isObject(release.composition)) {
+		composition = release.composition;
+	} else {
+		try {
+			composition = JSON.parse(release.composition);
```

Comment:
When does this ever need to be parsed?

Reply:
we use it further down
```
	return {
		releaseId: release.id,
		releaseVersion: release.release_version,
		commit: release.commit,
		name: application.app_name,
		services,
		networks: composition?.networks || {},
		volumes: composition?.volumes || {},
		appId: application.id,
		uuid: application.uuid,
	};
```

Reply:
I meant more specifically when is it ever not already parsed - I'd expect it to always be a parsed object at this point and not a string

Reply:
Although with the refactoring being split out it's clear this was just an existing wart so it can be left as-is, but I do expect it to be unnecessary and we can clean it up in future